lib/pliny/librato/metrics/backend.rb

```diff
@@ -54,23 +54,20 @@ def start_timer
           end
         end
 
-        def start_counter
-          @counter = Thread.new do
-            loop do
-              msg = metrics_queue.pop
-              msg == POISON_PILL ? break : enqueue_librato(msg)
+        def flush_librato
+          sync do
```

Comment:
Any time `flush_librato` is running the `sync` here has the side effect that `report_counts` and `report_measure` will block. The way we set this code up before was that reporting metrics was a non blocking and thread safe operation.